### PR TITLE
Jira/PE-54: Implement RESTful API base on PE-23

### PR DESCRIPTION
**Pull request by freddy61025**
https://issues.opennms.org/browse/PE-54

### CODE REVIEW

File: features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/ConfigManagerRestServiceImpl.java [hunk 1]
```diff
@@ -94,58 +95,79 @@ public Response getOpenApiSchema(String configName, String acceptType){
...
 
-    /**
-     * get or create a fake schema and return
-     *
-     * @param configName
-     * @return
-     */
     @Override
-    public ConfigSchema getSchema(String configName) {
+    public Response getConfig(String configName, String configId) {
```
Comment (patrick-schweizer) [inline features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/ConfigManagerRestServiceImpl.java:108]: I wonder if we should have a 404 if a config can't be found?

**Review by patrick-schweizer — COMMENTED** [review]
I have just a view minor comments about the code.
The big question is for me:
Do we want / need partial retrieving and storing of a configuration or if it is enough to always load and save the full config (as now implemented).

File: features/config/rest/api/src/main/java/org/opennms/features/config/rest/api/ConfigManagerRestService.java [hunk 2]
```diff
@@ -52,33 +48,74 @@
...
+    /**
+     *
+     * @param configName
+     * @param configId
+     * @param jsonStr
+     * @return
+     */
+    @PUT
+    @Path("/{configName}/{configId}")
+    Response updateConfig(@PathParam("configName") String configName, @PathParam("configId") String configId, String jsonStr);
```
[the remark this answers is not in the document]
Comment (freddy61025) [reply features/config/rest/api/src/main/java/org/opennms/features/config/rest/api/ConfigManagerRestService.java:108]: I have make it support partial update at ConfigStoreDaoImpl.updateConfig()

File: features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/ConfigManagerRestServiceImpl.java [hunk 3]
```diff
@@ -94,58 +95,79 @@ public Response getOpenApiSchema(String configName, String acceptType){
...
-                schema = configurationManagerService.getRegisteredSchema(configName);
-            }
-            return schema.get();
+            String jsonStr = configurationManagerService.getJSONStrConfiguration(configName, configId);
+            return Response.ok(jsonStr).build();
         } catch (Exception e) {
-            e.printStackTrace();
-            return null;
-//            return Response.serverError().build();
+            LOG.error("configName: " + configName + " configId:" + configId + " " + e.getMessage());
```
[the remark this answers is not in the document]
Comment (freddy61025) [reply features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/ConfigManagerRestServiceImpl.java]: done

File: features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/ConfigManagerRestServiceImpl.java (hunk 1 above)
Comment (freddy61025) [reply features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/ConfigManagerRestServiceImpl.java:108]: done

**Review by patrick-schweizer — APPROVED** [review]

**Pull request merged**